ocs_ci/deployment/aws.py

```diff
@@ -134,18 +152,247 @@ def destroy_cluster(self, log_level="DEBUG"):
+
+            with open("./upi_on_aws-install.sh", "r") as fd:
+                buf = fd.read()
+            data = buf.replace("openshift-qe-upi", "ocs-qe-upi")
+            with open("./upi_on_aws-install.sh", "w") as fd:
+                fd.write(data)
+
+            # ensure oc is on PATH for UPI install script
+            test_env = os.environ.copy()
+            path = test_env.get('PATH')
+            oc_path = os.path.join(constants.TOP_DIR, 'bin')
+            test_env['PATH'] = f'{oc_path}:{path}'
+
+            logger.info("Executing UPI install script")
+            proc = Popen(
+                [os.path.join(self.upi_script_path, 'upi_on_aws-install.sh')],
+                stdout=PIPE, stderr=PIPE, env=test_env
+            )
+            stdout, stderr = proc.communicate()
+
+            # Change dir back to ocs-ci dir
+            os.chdir(cidir)
+
+            if proc.returncode:
+                logger.error(stderr)
+                raise exceptions.CommandFailed("upi install script failed")
+            logger.info(stdout)
+
+            self.test_cluster()
+
+    def deploy_ocp(self, log_cli_level='DEBUG'):
+        """
+        OCP deployment specific to AWS UPI
+
+        Args:
+             log_cli_level (str): openshift installer's log level
+                (default: 'DEBUG')
+        """
+        super(AWSUPI, self).deploy_ocp(log_cli_level)
+        volume_size = config.ENV_DATA.get('DEFAULT_EBS_VOLUME_SIZE', 100)
+        # existing function looks for terraform files
+        self.add_volume(volume_size)
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Destroy OCP cluster for AWS UPI
+
+        Args:
+            log_level (str): log level for openshift-installer (
+                default:DEBUG)
+        """
+        cluster_name = get_cluster_name(self.cluster_path)
+        # Destroy extra volumes
+        self.destroy_volumes()
+
+        # Create cloudformation client
+        cf = boto3.client('cloudformation')
+
+        # Delete master, bootstrap, security group, and worker stacks
+        suffixes = ['ma', 'bs', 'sg']
+        # TODO: read in num_workers in a better way
+        num_workers = int(os.environ.get('num_workers', 3))
+        for i in range(num_workers - 1, -1, -1):
+            suffixes.insert(0, f'no{i}')
+        stack_names = [f'{cluster_name}-{suffix}' for suffix in suffixes]
+        for stack_name in stack_names:
+            logger.info("Destroying stack: %s", stack_name)
+            cf.delete_stack(StackName=stack_name)
+            verify_stack_deleted(stack_name)
+
+        # Call openshift-installer destroy cluster
+        super(AWSUPI, self).destroy_cluster(log_level)
+
+        # Delete inf and vpc stacks
+        suffixes = ['inf', 'vpc']
+        stack_names = [f'{cluster_name}-{suffix}' for suffix in suffixes]
+        for stack_name in stack_names:
+            logger.info("Destroying stack: %s", stack_name)
```

Comment:
Can you please check if it also delete s3 buckets? When I did today's cleanup I saw many of them are with UPI inside name, so worth to check if we don't miss something when destroying the cluster.

Reply:
I think the buckets that are still in s3 are due leftovers from testing, and not from our successful deployments/teardowns. I will verify that this is the case and if it's not look into a fix.

Reply:
Just ran a local test and no s3 buckets were left after teardown.

Reply:
cool, thx